src/main/java/org/opennars/inference/TruthFunctions.java

```diff
@@ -46,7 +46,7 @@ public static TruthValue dispatchBinary(final EnumType type, final TruthValue a,
         }
     }
 
-    public static TruthValue dispatchBinary2(final boolean v, final EnumType typeTrue, final EnumType typeFalse, final TruthValue a, final TruthValue b) {
+    public static TruthValue dispatchBinaryByBool(final boolean v, final EnumType typeTrue, final EnumType typeFalse, final TruthValue a, final TruthValue b) {
```

Comment:
I would say same thing here, use overloading... They Im not sure why youd want to distinguish them in this way when they functionally serve the same purpose just different arugments.

Reply:
They don't serve the same purpose - dispatchBinary is called binary because it has two arguments (the truth) - By bool decides the type based on the bool - it's just different